internal/compiler/compiler.go

```diff
@@ -875,6 +877,35 @@ func (c *compiler) expr(expr ast.Expr) {
 	}
 }
 
+// Generate a Concat opcode or, if possible compact multiple `Concat` into one `ConcatN`
+func (c *compiler) concatOp(e *ast.BinaryExpr) {
+	var parts []ast.Expr
+
+	for {
+		if be, ok := e.Left.(*ast.BinaryExpr); ok && be.Op == lexer.CONCAT {
+			parts = append(parts, e.Right)
+			e = be
+		} else {
+			parts = append(parts, e.Right)
+			parts = append(parts, e.Left)
+			break
+		}
+	}
```

Comment:
What about this structure (and variable naming)? IMO it's a bit clearer and avoids the `else` (the early-exit pattern is more common in Go).

```go
func (c *compiler) concatOp(expr *ast.BinaryExpr) {
	var values []ast.Expr
	for {
		values = append(values, expr.Right)
		left, isBinary := expr.Left.(*ast.BinaryExpr)
		if !isBinary || left.Op != lexer.CONCAT {
			break
		}
		expr = left
	}
	values = append(values, expr.Left)
        ...
```

Reply:
Yup! This looks nicer!